.github/workflows/template-maintenance.yml

```diff
@@ -0,0 +1,262 @@
+name: Template Maintenance
+
+on:
+  workflow_run:
+    workflows: ["Release Please"]
+    types:
+      - completed
+    branches:
+      - main
+
+# Prevent multiple simultaneous runs
+concurrency:
+  group: template-maintenance-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+  pull-requests: write
+  actions: read
+
+jobs:
+  detect-release:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    outputs:
+      should-upgrade: ${{ steps.check-release.outputs.should-upgrade }}
+      release-tag: ${{ steps.check-release.outputs.release-tag }}
+      release-type: ${{ steps.check-release.outputs.release-type }}
+      release-url: ${{ steps.check-release.outputs.release-url }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.TAMBO_UPGRADE_PAT || secrets.GITHUB_TOKEN }}
+
+      - name: Make detection script executable
+        run: chmod +x .github/scripts/detect-release.sh
```

Comment:
this should be marked executable within the repo, no need to chmod during the workflow